src/lpf2hub.ts

```diff
@@ -362,31 +362,45 @@ export class LPF2Hub extends Hub {
 
 
     private _parseModeInformationResponse (data: Buffer) {
-        const port = toHex(data[3]);
+        const portHex = toHex(data[3]);
         const mode = data[4];
         const type = data[5];
+        const port = this._getPortForPortNumber(data[3]) as Port;
+
+        if (!port) {
+            return false;
+        }
+
+        port.modes[mode] = port.modes[mode] || {};
```

Comment:
`port.modes` should be reset on device detach event. Otherwise the intoformation for a removable port will be a mix of old and new device data.